src/loaders/json.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { nest, cast } from '../util.js';
+
+function ispair(e) {
+  return Array.isArray(e) && e.length === 2 && typeof e[0] === 'string' && !Array.isArray(e[1]) && typeof e[1] !== 'object';
```

Comment:
- is the order important?
- if it's not an object, is is never an array
- `ispair` is a bit a weak description of what the function is doing. how about adding JSDOC?

Reply:
To make it more understandable, I'd change it to the following after the first two checks:
```
if (Array.isArray(e) && e.length === 2) {
   const [key, value] = e;
   ...
}
```